File: oauth2-server/src/main/java/a/gleb/keycloaktojwt/db/entity/AccountRole.java
package a.gleb.keycloaktojwt.db.entity;


import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

import static jakarta.persistence.GenerationType.AUTO;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "account_role")
public class AccountRole {

    @Id
    @GeneratedValue(strategy = AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;

    @Column(name = "role_name")
    private String roleName;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "account_to_role",
            joinColumns = { @JoinColumn(name = "role_id", referencedColumnName = "id") },
            inverseJoinColumns = { @JoinColumn(name = "account_id") })
    private Set<Account> roles = new HashSet<>();
}

File: oauth2-server/src/main/kotlin/a/gleb/oauth2server/db/entity/AccountRole.kt
package a.gleb.oauth2server.db.entity

import jakarta.persistence.*
import jakarta.persistence.GenerationType.AUTO
import java.util.*

@Entity(name = "account_role")
data class AccountRole(
    @Id
    @GeneratedValue(strategy = AUTO)
    val id: UUID?,

    @Column(name = "role_name")
    val roleName: String,

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "account_to_role",
        joinColumns = [JoinColumn(name = "role_id", referencedColumnName = "id")],
        inverseJoinColumns = [JoinColumn(name = "account_id")]
    )
    private val roles: Set<Account> = HashSet()

)